### PR TITLE
chore!(config, docs): delete deprecated vector.toml code

### PR DESCRIPTION
**Pull request by pront**
closes: #18323

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]

File: src/cli.rs [hunk 1]
```diff
@@ -64,9 +65,7 @@ pub struct RootOpts {
     /// Read configuration from one or more files. Wildcard paths are supported.
     /// File format is detected from the file name.
     /// If zero files are specified the deprecated default config path
```
Comment (hestonhoffman) [inline src/cli.rs:67]:
```suggestion
    /// If zero files are specified, the deprecated default config path
```

File: src/cli.rs [hunk 2]
```diff
@@ -64,9 +65,7 @@ pub struct RootOpts {
...
-    /// `/etc/vector/vector.toml` will be targeted.
-    /// And if the aforementioned file does not exist,
-    /// then `/etc/vector/vector.yaml` will be used.
+    /// `/etc/vector/vector.yaml` will be targeted.
```
Comment (hestonhoffman) [inline src/cli.rs:68]:
```suggestion
    /// `/etc/vector/vector.yaml` is targeted.
```

File: src/config/cmd.rs [hunk 3]
```diff
@@ -21,9 +21,7 @@ pub struct Opts {
     /// Read configuration from one or more files. Wildcard paths are supported.
     /// File format is detected from the file name.
     /// If zero files are specified the deprecated default config path
```
Comment (hestonhoffman) [inline src/config/cmd.rs:23]:
```suggestion
    /// If zero files are specified, the deprecated default config path
```

File: src/config/cmd.rs [hunk 4]
```diff
@@ -21,9 +21,7 @@ pub struct Opts {
...
-    /// `/etc/vector/vector.toml` will be targeted.
-    /// And if the aforementioned file does not exist,
-    /// then `/etc/vector/vector.yaml` will be used.
+    /// `/etc/vector/vector.yaml` will be targeted.
```
Comment (hestonhoffman) [inline src/config/cmd.rs:24]:
```suggestion
    /// `/etc/vector/vector.yaml` is targeted.
```

**Review by hestonhoffman — APPROVED** [review]
Left a couple small edits

**Pull request merged**